integration_test/tests/power_bi_tests.rs

```diff
@@ -9,11 +9,54 @@ mod integration {
         AttrConnectionPooling, AttrOdbcVersion, ConnectionAttribute, DriverConnectOption,
         EnvironmentAttribute, HDbc, HEnv, Handle, HandleType, InfoType, Pointer, SQLAllocHandle,
         SQLDriverConnectW, SQLFreeHandle, SQLGetInfoW, SQLSetConnectAttrW, SQLSetEnvAttr, SmallInt,
-        SqlReturn,
+        SqlReturn, USmallInt,
     };
     use std::ptr::null_mut;
     use std::slice;
 
+    macro_rules! test_get_info {
+        ($conn_handle:expr, $env_handle: expr, $info_type: expr , $actual_value_modifier: ident) => {{
+            let conn_handle = $conn_handle;
+            let env_handle = $env_handle;
+            let info_type = $info_type;
+            let actual_value_modifier = $actual_value_modifier;
+
+            let str_len_ptr = &mut 0;
+            const BUFFER_LENGTH: SmallInt = 300;
+            let output_buffer = &mut [0u16; (BUFFER_LENGTH as usize - 1)] as *mut _;
+
+            let outcome = SQLGetInfoW(
+                conn_handle as HDbc,
+                info_type,
+                output_buffer as Pointer,
+                BUFFER_LENGTH,
+                str_len_ptr,
+            );
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                outcome,
+                "Expected {}, got {}. Diagnostic message is: {}",
+                sql_return_to_string(SqlReturn::SUCCESS),
+                sql_return_to_string(outcome),
+                get_sql_diagnostics(HandleType::Env, env_handle as Handle)
+            );
+            println!(
+                "{:?} = {}\nLength is {}",
+                info_type,
+                actual_value_modifier(output_buffer as Pointer, *str_len_ptr as usize),
+                *str_len_ptr
+            );
+        }};
+    }
+
+    unsafe fn modify_string_value(value_ptr: Pointer, out_length: usize) -> String {
+        String::from_utf16_lossy(slice::from_raw_parts(value_ptr as *const _, out_length))
+    }
+
+    unsafe fn modify_u16_value(value_ptr: Pointer, _: usize) -> u16 {
```

Comment:
Same here, I don't really understand the prefix.
I think you mean to say that it's reading/converting the content of the output buffer into a u16 (USmallint). It is not modifying anything as far as I can tell though.